cl/corpus_importer/tasks.py

```diff
@@ -1570,11 +1574,79 @@ def download_pacer_pdf_by_rd(
     s = PacerSession(cookies=cookies)
     report = FreeOpinionReport(pacer_court_id, s)
 
-    r, r_msg = report.download_pdf(pacer_case_id, pacer_doc_id, magic_number)
+    r, r_msg = report.download_pdf(
+        pacer_case_id, pacer_doc_id, magic_number, appellate
+    )
 
     return r, r_msg
 
 
+def get_document_number_from_confirmation_page(
+    court_id: str, pacer_doc_id: str
+) -> str:
+    """Get the PACER document number from the PACER download confirmation page.
+
+    :param court_id: A CourtListener court ID to query the confirmation page.
+    :param pacer_doc_id: The pacer_doc_id to query the confirmation page.
+    :return: The PACER document number is available or an empty string if not.
+    """
+
+    recap_email_user = User.objects.get(username="recap-email")
+    cookies = get_or_cache_pacer_cookies(
+        recap_email_user.pk, settings.PACER_USERNAME, settings.PACER_PASSWORD
+    )
+    s = PacerSession(cookies=cookies)
+    doc_num_report = DownloadConfirmationPage(court_id, s)
+    doc_num_report.query(pacer_doc_id)
+    data = doc_num_report.data
+    return data.get("document_number", "")
+
+
+def get_document_number_appellate(
+    court_id: str, pacer_doc_id: str, response: Response | None
```

Comment:
I think I'd just pass response.content here so that your signature becomes:

```suggestion
    court_id: str, pacer_doc_id: str, pdf_bytes: bytes | None
```

That should make testing easier and make this function more clear. When you send the whole response, it implies you have a reason to do so, but we just need the bytes, so we should just send only that.

Reply:
This was refactorized now this function doesn't receive the PDF Response. Now the PDF is requested inside this function since we need to get the PDF before adding the docket entry in order to get the document number.